scripts/lib/CIME/SystemTests/irt.py

```diff
@@ -0,0 +1,56 @@
+"""
+Implementation of the CIME IRT. (Interum Restart Test)
+This test the model restart capability as well as the short term archivers interum restart capabilty
+
+(1) Do a Run of length N with restart at N/2 and SAVE_INTERUM_RESTART set to TRUE
+(2) Archive Run using ST archive tools
+(3) Recover first interum restart to the case2 run directory
+(4) Start case2 from restart and run to the end of case1
+(5) compare results.
+
+"""
+
+from CIME.SystemTests.system_tests_compare_two import SystemTestsCompareTwo
+from CIME.XML.standard_module_setup import *
+from CIME.case_setup import case_setup
+from CIME.case_st_archive import case_st_archive, restore_from_archive
+from CIME.utils import sorted_ls
+
+logger = logging.getLogger(__name__)
+
+class IRT(SystemTestsCompareTwo):
+
+    def __init__(self, case):
+        SystemTestsCompareTwo.__init__(self, case,
+                                       separate_builds=False,
+                                       run_two_suffix = 'restart',
+                                       run_one_description = 'initial',
+                                       run_two_description = 'restart',
+                                       multisubmit = False)
+
+    def _case_one_setup(self):
+        case_setup(self._case)
```

Comment:
Why is `case_setup` needed here?

Also, as in ers and similar tests, we should have an `expect` that ensures an appropriate run length has been chosen. i.e., stop_n needs to be at least 3.